app/demo-interactor/KNDemoKernel.cu

```diff
@@ -0,0 +1,191 @@
+        {
+            // Particle is below interaction energy; kill and deposit energy
+            energy_deposition[tid] = particle.energy();
+            alive[tid]             = false;
+            continue;
+        }
+
+        // Construct RNG and interaction interfaces
+        RngEngine              rng(rng_states, ThreadId(tid));
+        KleinNishinaInteractor interact(
+            kn_params, particle, direction[tid], allocate_secondaries);
+
+        // Perform interaction: should emit a single particle (an electron)
+        auto interaction = interact(rng);
+        CHECK(interaction);
+        CHECK(interaction.secondaries.size() == 1);
+
+        // Deposit energy from the secondary (effectively, an infinite energy
+        // cutoff)
+        energy_deposition[tid] = interaction.secondaries.front().energy;
+
+        // Update post-interaction state (apply interaction)
+        direction[tid] = interaction.direction;
+        particle.energy(interaction.energy);
+    }
+}
+
+//---------------------------------------------------------------------------//
+// HOST INTERFACES
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize particle states.
+ */
+void initialize(CudaGridParams         grid,
+                ParticleParamsPointers params,
+                ParticleStatePointers  states,
+                ParticleTrackState     initial_state,
+                RngStatePointers       rng_states,
+                span<Real3>            direction,
+                span<bool>             alive)
+{
+    REQUIRE(alive.size() == rng_states.rng.size());
+    REQUIRE(alive.size() == states.vars.size());
+    initialize_kn<<<grid.grid_size, grid.block_size>>>(
+        params, states, initial_state, direction.data(), alive.data());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Run an iteration.
+ */
+void iterate(CudaGridParams                 kernel_params,
+             ParticleParamsPointers         particle_params,
+             ParticleStatePointers          particle_states,
+             KleinNishinaInteractorPointers kn_params,
+             SecondaryAllocatorPointers     secondaries,
+             RngStatePointers               rng_states,
+             span<Real3>                    direction,
+             span<bool>                     alive,
+             span<real_type>                energy_deposition)
+{
+    iterate_kn<<<kernel_params.grid_size, kernel_params.block_size>>>(
+        particle_params,
+        particle_states,
+        kn_params,
+        secondaries,
+        rng_states,
+        direction.data(),
+        alive.data(),
+        energy_deposition.data());
+
+    CELER_CUDA_CALL(cudaDeviceSynchronize());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sum the total energy deposition.
+ */
+real_type reduce_energy_dep(span<real_type> edep)
```

Comment:
If I remember correctly, the relevance of this (and energy deposition) was discussed in detail in the physics meeting today.